### DISCUSSION TITLE
Handle unsuccessful request

### DISCUSSION THREAD

**Discussion by huesla** [grafana/grafana-json-datasource#259] [Q&A]
Hello, 

we are currently working on a Grafana dashboard that should display the healthstate of various backends with different colors. It can happen that the backends are not reachable, which leads to the panel displaying an error symbol and "No Data".

![2022-02-02 17_25_59-Window](https://user-images.githubusercontent.com/81033366/152195809-885fc4fe-d663-485a-8318-4ae7c269fb09.png)

We would like to offer something more meaningful for the user. Is there an option to handle unsuccessful requests differently?

**Comment by Mario-Hofstaetter** [comment]
If you hover over the ❗ exclamation mark in the top left corner, there should be a tooltip with more information, what does it say?

**Comment by huesla** [comment]
![image](https://user-images.githubusercontent.com/81033366/152821254-3e72cba8-e713-436e-ba9b-0c6864bad864.png)

and under "Query inspector" -> "Error":
```
Bad Gateway
[Object]()
[status:502]()
[statusText:"Bad Gateway"]()
[data:Object]()
[message:"Bad Gateway"]()
[error:"Bad Gateway"]()
[response:""]()
[config:Object]()
[url:"api/datasources/proxy/4/api/insights/lifters?api-version=1.0"]()
[method:"GET"]()
[headers:Object]()
[retry:0]()
[hideFromInspector:false]()
[message:"Bad Gateway"]()
```

**Comment by huesla** [comment]
The users of this dashboard will not have the technical skill to understand "Bad Gateway", therefore we need to show a customized error message.
